### PR TITLE
gobject-introspection: upgrade 1.71.0->1.79.1; meson: upgrade 1.2.1->1.3.1; glib: upgrade 2.78.1->2.79.1

### PR DESCRIPTION
**Pull request by BettyRain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?

###### Change Log  

- gobject-introspection: upgrade 1.71.0->1.79.1; meson: upgrade 1.2.1->1.3.1; glib: upgrade 2.78.1->2.79.1
- Skip some meson tests because of their flakiness or we don't support them
- Add python-packaging to build_official_toolchain_rpms as required for new glib version

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- [Buddy Build Id](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=515762&view=results)
- [Full Build Id](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=515763&view=results)

### CODE REVIEW

**Comment by BettyRain** [comment]
Looks like toolchain build is failing for me because of python3-packaging-23.2-2.azl3.noarch.rpm
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=515772&view=logs&j=db98f19e-da46-5e4d-a4ba-372cf3771a92&t=8831c5ce-9476-5d69-9bf5-bfd7485723f1&l=464717

File: SPECS/gobject-introspection/gobject-introspection.spec [hunk 1]
```diff
@@ -98,6 +98,10 @@ find %{buildroot} -type f -name "*.la" -delete -print
 %{_mandir}/man1/*.gz
 
 %changelog
+%changelog
```
Comment (anphel31) [inline SPECS/gobject-introspection/gobject-introspection.spec:101]: duplicate %changelog can be removed

File: SPECS/meson/meson.spec [hunk 2]
```diff
@@ -57,6 +69,10 @@ python3 ./run_tests.py
 %{_datadir}/polkit-1/actions/com.mesonbuild.install.policy
 
 %changelog
+* Tue Feb 27 2024 Betty Lakes <bettylakes@microsoft.com> - 1.3.1-1
+- Update version to 1.3.0
```
Comment (anphel31) [inline SPECS/meson/meson.spec:73]: version 1.3.1

File: SPECS/glib/glib.spec [hunk 3]
```diff
@@ -120,9 +124,11 @@ touch %{buildroot}%{_libdir}/gio/modules/giomodule.cache
 %{_datadir}/glib-2.0/schemas/*
 
 %files doc
```
Comment (anphel31) [inline SPECS/glib/glib.spec:126]: if there are no docs, the entire subpackage could be removed

**Pull request closed without merging**